resolver/src/hosts.rs

```diff
@@ -0,0 +1,121 @@
+use std::collections::HashMap;
+use std::fs;
+use std::time::{Duration, SystemTime};
+use std::io::{BufRead, BufReader};
+use std::fs::File;
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+use std::sync::Mutex;
+
+lazy_static! {
+    static ref CACHE_MAX_AGE: Duration = Duration::new(5, 0);
+    static ref HOSTS: Mutex<Hosts> = Mutex::new(Hosts::new());
```

Comment:
One thing to consider is making this a ThreadLocal rather than a Mutex. This would simplify the threading considerations inside Tokio.